File: template.rb
#!/usr/bin/ruby

# = %%FILE%%
#
# Author::    %%USER%%
# Copyright:: %%USER%%
# License::   GPL
#

# == Version
#
# * %%DATE%% 1.00 (%%USER%%) 
#

# == Summary
# 
# This file is for
#

# == Interface
#
# * input
# **
# * output
# **
#

# == Example
#

# == Description
#
# This file is for
#

#require 'optparse'

#opt = OptionParser.new

#opt.banner  = <<EOF
#Usage : #{$0} [options] [filename]
#
#ex) #{$0} ./hoge.txt
#    cat hoge.txt | #{$0}
#
#this script is...
#
#[options]
#EOF

#begin

#  opt.on( '-a', 'hogehoge' ) { |v| p v }
#  opt.on( '-h', 'show helps and exit' ) { |v| 
#    puts opt
#    exit
#  }
#  opt.parse!( ARGV )
#  p ARGV

#rescue OptionParser::ParseError => e

#  puts opt
#  exit

#end

#if ARGV.size == 0 then
#  f = STDIN
#elsif ARGV.size == 1 then
#  f = File.open( ARGV[ 0 ] )
#  IN_FILE = ARGV[ 0 ]
#else
#  puts opt
#  exit
#end

#f.each do |line|
#  puts line
#end

File: template.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <getopt.h>

using namespace std ;

/*
 * %%FILE%%
 *
 *   Version 1.0 %%DATE%% %%USER%%
 *
 * this program is...
 *
 */

/*
void usage( char **argv ) {

  cout << "usage : " << argv[ 0 ] << "[options] [input file]" << endl ;
  cout << endl ;
  cout << "ex) " << argv[ 0 ] << " hoge.txt" << endl ;
  cout << "ex) cat hoge.txt | " << argv[ 0 ] << endl ;
  cout << endl ;
  cout << "[options]" << endl ;
  cout << " -h : show help and exit." << endl ;

}

void handle( istream &is ) {

  string buf ;

  while( getline( is, buf ) ) {
    cout << buf << endl ;
  }

}

int main( int argc, char **argv ) {

  int opt ;

  while( ( opt = getopt( argc, argv, "h" ) ) != -1 ) {
    switch( opt ) {
      case 'h' :
        usage( argv ) ;
        return 0 ;
      default :
        usage( argv ) ;
        return -1 ;
    }
  }

  if( argc == 1 ) {
    handle( cin ) ;
  } else if( argc == 2 ) {
    ifstream is( argv[ 1 ] ) ;
    if( ! is ) {
      cout << "failed to file open." << endl ;
      return -1 ;
    }
    handle( is ) ;
  } else {
    usage( argv ) ;
    return -1 ;
  }

  return 0 ;

}
*/

